src/controllers/training.ts

```diff
@@ -0,0 +1,18 @@
+import childProcess from 'child_process';
+import { Request, Response } from 'express';
+
+const getCommitHash = (): string => {
+  return childProcess.execSync('git rev-parse HEAD').toString().trim();
+};
+
+const getVersionController = async (req: Request, res: Response) => {
+  const commitHash: string = getCommitHash();
+  const version: string | undefined = process.env.npm_package_version;
```

Comment:
it should comes from package.json file.

Reply:
Corrected. I split this up into two separate typescript sub-projects. I wanted to use the ts 'import' syntax for *.json files.
With one TS configuration, the directory structure in output changed. Because package.json was sitting a level higher.